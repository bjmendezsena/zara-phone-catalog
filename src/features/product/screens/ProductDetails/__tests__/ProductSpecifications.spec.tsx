import { render, getHtml, mockProductDetailsResponse } from "@/tests";
import {
  ProductSpecifications,
  ProductSpecificationsProps,
} from "@/features/product/screens/ProductDetails/ProductSpecifications";

describe(`<${ProductSpecifications.name}/>`, () => {
  const defaultProps: ProductSpecificationsProps = {
    product: mockProductDetailsResponse,
  };

  it("renders without crashing", () => {
    const { container } = render(<ProductSpecifications {...defaultProps} />);
    expect(getHtml(container)).toMatchSnapshot();
  });

  it("displays the correct product specifications", () => {
    const { getByText } = render(<ProductSpecifications {...defaultProps} />);

    const specifications = [
      { label: "BRAND", value: mockProductDetailsResponse.brand },
      { label: "NAME", value: mockProductDetailsResponse.name },
      { label: "DESCRIPTION", value: mockProductDetailsResponse.description },
      { label: "SCREEN", value: mockProductDetailsResponse.specs.screen },
      {
        label: "RESOLUTION",
        value: mockProductDetailsResponse.specs.resolution,
      },
      { label: "PROCESSOR", value: mockProductDetailsResponse.specs.processor },
      {
        label: "MAIN CAMERA",
        value: mockProductDetailsResponse.specs.mainCamera,
      },
      {
        label: "SELFIE CAMERA",
        value: mockProductDetailsResponse.specs.selfieCamera,
      },
      { label: "BATTERY", value: mockProductDetailsResponse.specs.battery },
      { label: "OS", value: mockProductDetailsResponse.specs.os },
      {
        label: "SCREEN REFRESH RATE",
        value: mockProductDetailsResponse.specs.screenRefreshRate,
      },
    ];
    specifications.forEach((spec) => {
      expect(getByText(spec.label)).toBeInTheDocument();
      expect(getByText(spec.value)).toBeInTheDocument();
    });
  });

  it("returns null if no product is provided", () => {
    const { container } = render(<ProductSpecifications />);
    expect(container.firstChild).toBeNull();
  });
});
