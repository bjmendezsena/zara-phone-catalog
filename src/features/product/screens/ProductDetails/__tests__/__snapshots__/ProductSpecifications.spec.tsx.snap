// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ProductSpecifications/> renders without crashing 1`] = `
"<div>
  <div
    class="mt-4 xl:px-[360px] px-4"
  >
    <h2
      class="text-xl font-light text-primary mb-8 uppercase tracking-wide"
    >
      SPECIFICATIONS
    </h2>
    <div
      class="space-y-1"
    >
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          BRAND
        </div>
        <div
          class="text-sm"
        >
          Google
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          NAME
        </div>
        <div
          class="text-sm"
        >
          Pixel 8a
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          DESCRIPTION
        </div>
        <div
          class="text-sm"
        >
          Descubre Pixel 8a, creado por Google. Saca fotos magníficas con la Cámara Pixel. Haz más en menos tiempo con la IA de Google, como arreglar imágenes borrosas, filtrar llamadas y aprender cosas nuevas. Sus funciones de seguridad del más alto nivel ayudan a proteger tus datos. Y se ha diseñado para durar. Todo, a un precio excepcional.
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          SCREEN
        </div>
        <div
          class="text-sm"
        >
          6.1" OLED Actua
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          RESOLUTION
        </div>
        <div
          class="text-sm"
        >
          2400 x 1080 pixels
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          PROCESSOR
        </div>
        <div
          class="text-sm"
        >
          Google Tensor G3
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          MAIN CAMERA
        </div>
        <div
          class="text-sm"
        >
          64 MP + 13 MP
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          SELFIE CAMERA
        </div>
        <div
          class="text-sm"
        >
          13 MP
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          BATTERY
        </div>
        <div
          class="text-sm"
        >
          4492 mAh
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          OS
        </div>
        <div
          class="text-sm"
        >
          Android
        </div>
      </div>
      <div
        class="grid grid-cols-2 gap-4 py-4 border-b border-primary first:border-t"
      >
        <div
          class="text-sm uppercase tracking-wide font-medium"
        >
          SCREEN REFRESH RATE
        </div>
        <div
          class="text-sm"
        >
          120 Hz
        </div>
      </div>
    </div>
  </div>
</div>"
`;
